relayer/relays/parachain/beefy-listener.go

```diff
@@ -696,26 +696,32 @@ func (li *BeefyListener) scanForCommitments(
 				if bundleNonce < incentivizedNonceToFind {
 					scanIncentivizedChannelDone = true
 					continue
-					// Collect these commitments
 				} else if bundleNonce > incentivizedNonceToFind {
-					commitments[channelID] = append(commitments[channelID], NewCommitment(digestItem.AsCommitment.Hash, bundle))
-					// collect this commitment and terminate scan
+					// Collect these commitments
+					commitment := NewIncentivizedChannelCommitment(digestItem.AsCommitment.Hash, bundle)
+					incentivizedChannelCommitment = &commitment
```

Comment:
Perhaps `NewIncentivizedChannelCommitment` can return a reference to `IncentivizedChannelCommitment`, since it seems like it is only used here and in line `705`? That would remove the need for the extra line on `702` and `706`.

Reply:
Will do. I was trying to stick to values, especially for a constructor function like this. But this is the only place it's used and we need a pointer for `Task` anyway